src/libostree/ostree-sysroot.c

```diff
@@ -1108,6 +1108,38 @@ find_booted_deployment (OstreeSysroot       *self,
 }
 
 /**
+ * ostree_sysroot_get_last_deployed:
+ * @self: Sysroot
+ * @osname: (allow-none): Operating system group
+ *
+ * Find the last deployed sysroot.
+ *
+ * Returns: (transfer full): The last deployment in OS or NULL if none
+ */
+OstreeDeployment *
+ostree_sysroot_get_last_deployed (OstreeSysroot     *self,
```

Comment:
Makes sense to me, can you try that @OYTIS  ?  `ostree_sysroot_query_deployments` ?